src/Aspire.Hosting.Azure/Bicep/AzureBicepServiceBusResource.cs

```diff
@@ -29,6 +34,63 @@ public class AzureBicepServiceBusResource(string name) :
     public string? GetConnectionString() => ServiceBusEndpoint.Value;
 }
 
+/// <summary>
+/// Represents an Azure Service Bus Queue resource.
+/// </summary>
+public class AzureBicepServiceBusQueueResource(string name, AzureBicepServiceBusResource parent) :
+    Resource(name),
+    IResourceWithConnectionString,
+    IResourceWithParent<AzureBicepServiceBusResource>
+{
+    /// <inheritdoc/>
+    public AzureBicepServiceBusResource Parent { get; } = parent;
+
+    /// <inheritdoc/>
+    public string ConnectionStringExpression => $"{{{Parent.Name}.connectionString}}";
+
+    /// <inheritdoc/>
+    public string? GetConnectionString() => Parent.GetConnectionString();
+
+    internal void WriteToManifest(ManifestPublishingContext context)
+    {
+        // REVIEW: What do we do with resources that are defined in the parent's bicep file?
+        context.Writer.WriteString("type", "azure.bicep.v0");
+        context.Writer.WriteString("connectionString", ConnectionStringExpression);
+        context.Writer.WriteString("parent", Parent.Name);
+    }
+}
+
+/// <summary>
+/// Represents an Azure Service Bus Queue resource.
+/// </summary>
+public class AzureBicepServiceBusTopicResource(string name, string[] subscriptions, AzureBicepServiceBusResource parent) :
+    Resource(name),
+    IResourceWithConnectionString,
```

Comment:
Should this really implement `IResourceWithConnectionString`? The connection string is on the namespace level.

Reply:
I'm not sure this should be a resource either.